modules/config/i18n.nix

```diff
@@ -0,0 +1,44 @@
+# The glibc package in nixpkgs is patched to make it possbile to specify
+# an alternative path for the locale archive through a special environment
+# variable. This would allow different versions of glibc to coexist on the
+# same system because each version of glibc could look up different paths
+# for its locale archive should the archive format ever change in
+# incompatible ways.
+#
+# See also:
+# localedef(1)
+# https://nixos.org/manual/nixpkgs/stable/#locales
+# https://github.com/NixOS/nixpkgs/issues/38991
+#
+# XXX: The name of the said environment variable gets updated with each
+# breaking release of the glibcLocales package. Periodically check the link
+# below for changes:
+# https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/development/libraries/glibc/nix-locale-archive.patch
+
+{ lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (pkgs.glibcLocales) version;
+
+  archivePath = "${pkgs.glibcLocales}/lib/locale/locale-archive";
+
+  # lookup the version of glibcLocales and set the appropriate environment vars
+  localeVars = if (versionAtLeast version "2.27") then {
+    LOCALE_ARCHIVE_2_27 = archivePath;
+  } else if (versionAtLeast version "2.11") then {
+    LOCALE_ARCHIVE_2_11 = archivePath;
```

Comment:
I'm not sure if this would alleviate your concern, but the name of the environment variable `LOCALE_ARCHIVE_X_XX` gets updated only when the format of the locale archive is changed upstream in glibc. Since there are no incompatibilities between 2.11 and 2.16, they all use `LOCALE_ARCHIVE_2_11`.

Reply:
Thanks for the explanation :+1: